Microsoft.Toolkit.Uwp.UI.Controls/TokenizingTextBox/TokenizingTextBox.cs

```diff
@@ -73,11 +73,26 @@ protected override void OnApplyTemplate()
             ContextFlyout = menuFlyout;
         }
 
-        private void AutoSuggestBox_KeyDown(object sender, KeyRoutedEventArgs e)
+        private async void AutoSuggestBox_KeyDown(object sender, KeyRoutedEventArgs e)
         {
             switch (e.Key)
             {
                 case VirtualKey.Back:
+                    {
+                        TextBox autoSuggestTextBox = _autoSuggestBox.FindDescendant<TextBox>() as TextBox;
+                        if (autoSuggestTextBox != null)
+                        {
+                            int currentCursorPosition = autoSuggestTextBox.SelectionStart;
+                            if (currentCursorPosition == 0)
```

Comment:
To address your concern you mentioned about distinguishing between moving from index 1->0 vs. 0->0, could the [previewkeydown](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.previewkeydownevent#Windows_UI_Xaml_UIElement_PreviewKeyDownEvent), [beforetextchanging](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textbox.beforetextchanging), or [selectionchanging](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textbox.selectionchanging) events help us so we can record the selectionstart index at that point to compare and then only move if they're both 0?